paddle/framework/net.h

```diff
@@ -55,43 +52,22 @@ class Operator {
  * This is the base class of network, all the networks should implement the apis
  * it defines.
  */
-class Net {
+class Net : public OperatorBase {
  public:
   /**
-   * @brief Infer shapes of all inputs and outputs of operators.
-   */
-  virtual void InferShape(Scope *scope) = 0;
-  /**
-   * @brief Run the network.
-   *
-   * Run all the operators and return success(true) or not, with all the
-   * variables are located in `scope`. `context` describes the detail execution
-   * environment for ops. `begin` and `end` specify the scope of `ops_` to run,
-   * If no positive indexes are provided, all operators in `ops_` will run.
-   */
-  virtual void Run(std::shared_ptr<Scope> scope, DeviceContext *ctx) = 0;
-
-  /**
    * @brief Add an Operator according to `def`.
    */
-  virtual OpIndex AddOp(const OpProto &def) = 0;
+  virtual void AddOp(const OpDesc& def) = 0;
```

Comment:
I think it is better to make `OpCreator` returns a `std::shared_ptr<OpBase>`, and we could alias it to `OpPtr`.

`AddOp` should take a `OpPtr`.